.ci.yml

```diff
@@ -0,0 +1,267 @@
+        echo "Error: $FAIL_MSG"
+        echo -e "TAP version 13\n1..1\nnot ok 1 $FAIL_MSG\n  ---\n  duration_ms: 0\n  ..." > withoutintl-test.tap
+        exit 1
+      fi
+
+  withoutssl:
+    label: Linux --without-ssl
+    image: ubuntu1804
+    execute: |
+      export CONFIG_FLAGS="$CONFIG_FLAGS --without-ssl"
+      make run-ci -j $JOBS V=
+
+      HAS_OPENSSL="$(out/Release/node -p 'Boolean(process.versions.openssl)')"
+      echo "Has OpenSSL: $HAS_OPENSSL"
+      if [ X"$HAS_OPENSSL" != X"false" ]; then
+        FAIL_MSG="Has an OpenSSL"
+        echo "Error: $FAIL_MSG"
+        echo -e "TAP version 13\n1..1\nnot ok 1 $FAIL_MSG\n  ---\n  duration_ms: 0\n  ..." > withoutssl-test.tap
+        exit 1
+      fi
+      out/Release/node -p 'require("crypto")' || REQUIRE_CRYPTO="no crypto"
+      if [ "$REQUIRE_CRYPTO" != "no crypto" ]; then
+        FAIL_MSG='require("crypto") did not fail'
+        echo "Error: $FAIL_MSG"
+        echo -e "TAP version 13\n1..1\nnot ok 1 $FAIL_MSG\n  ---\n  duration_ms: 0\n  ..." > withoutssl-test.tap
+        exit 1
+      fi
+
+  shared:
+    label: Linux --shared
+    image: ubuntu1804
+    execute: |
+      export CONFIG_FLAGS="$CONFIG_FLAGS --shared"
+      make run-ci -j $JOBS V=
+
+  debug:
+    label: Linux Debug
+    image: ubuntu1804
+    execute: |
+      export CONFIG_FLAGS="$CONFIG_FLAGS --debug"
+
+      # see https://github.com/nodejs/node/issues/17016
+      sed -i 's/\[\$system==linux\]/[$system==linux]\ntest-error-reporting : PASS, FLAKY/g' test/parallel/parallel.status
+      # see https://github.com/nodejs/node/issues/17017
+      sed -i 's/\[\$system==linux\]/[$system==linux]\ntest-inspector-async-stack-traces-promise-then : PASS, FLAKY/g' test/sequential/sequential.status
+      # see https://github.com/nodejs/node/issues/17018
+      sed -i 's/\[\$system==linux\]/[$system==linux]\ntest-inspector-contexts : PASS, FLAKY/g' test/sequential/sequential.status
+
+      make build-ci -j $JOBS V=
+
+      if ! [ -x out/Debug/node ]; then
+        FAIL_MSG="No Debug executable"
+        echo "Error: $FAIL_MSG"
+        echo -e "TAP version 13\n1..1\nnot ok 1 $FAIL_MSG\n  ---\n  duration_ms: 0\n  ..." > debug-test.tap
+        exit 1
+      fi
+
+      BUILD_TYPE="$(out/Debug/node -pe process.config.target_defaults.default_configuration)"
+      echo "Build type: $BUILD_TYPE"
+      if [ X"$BUILD_TYPE" != X"Debug" ]; then
+        FAIL_MSG="Not built as Debug"
+        echo "Error: $FAIL_MSG"
+        echo -e "TAP version 13\n1..1\nnot ok 1 $FAIL_MSG\n  ---\n  duration_ms: 0\n  ..." > debug-test.tap
+        exit 1
+      fi
+
+      python tools/test.py -j $JOBS -p tap --logfile test.tap \
+        --mode=debug --flaky-tests=$FLAKY_TESTS \
+        async-hooks default known_issues
+
+  ninja:
+    label: Linux Ninja
+    image: ubuntu1804
+    execute: |
+      export CC=/usr/lib/ccache/cc
+      export CXX=/usr/lib/ccache/c++
+      python ./configure --verbose --ninja
```

Comment:
Is ninja in first-tier support ? Maybe also just check the config part ?

Reply:
Not "first-tier", but we keep on getting complaints when it breaks so I figure it might be good to test, can be removed if folks disagree - this proposal is more of a demonstration and I'm hoping others will expand it.
Regarding just the config part, I'm not sure what that would prove without validating the ninja build descriptor. The ninja built itself, when hooked up to ccache is quicker than classic `make`. It does take resources to run it so maybe it could be first to go if we ran into bottlenecks.